src/main/java/org/cactoos/iterator/CheckedDigit.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.cactoos.iterator;
+
+import java.io.IOException;
+import java.util.Iterator;
+import org.cactoos.text.FormattedText;
+
+/**
+ * Wrap around {@link Character#digit(char, int)}.
+ *
+ * @author Alexander Menshikov (sharplermc@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class CheckedDigit implements Iterator<Integer> {
```

Comment:
You are making it harder on yourself and duplicating huge amount of code. You can assemble this class out of reusable building blocks [IterableOf], [Mapped] and [IterableEnvelop] as I explained in https://github.com/yegor256/cactoos/pull/729#discussion_r176245736

If you don't know how to use Decorating Envelopes read this article: http://www.yegor256.com/2017/01/31/decorating-envelopes.html

[IterableOf]: https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/iterable/IterableOf.java
[Mapped]: https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/iterable/Mapped.java
[IterableEnvelop]: https://github.com/yegor256/cactoos/blob/63f3b7fcd7be411d573c8ef3ac99bdb10fa4b144/src/main/java/org/cactoos/iterable/IterableEnvelope.java

Reply:
Okay. I did that because `Iterator.next` can't throw `IOException`. And I'm trying to avoid casting between `IOException` and `UncheckedIOException` or throwing 2 type of exceptions.